packages/gatsby/src/schema/lazy-fields.js

```diff
@@ -25,24 +25,24 @@ const { GraphQLList, GraphQLObjectType } = require(`graphql`)
 // sift, rather than loki, so not a big deal
 const typeFields = new Map()
 
-function contains(filters, queryType, fieldType = queryType) {
+function contains(filters, fieldType) {
   return _.some(filters, (fieldFilter, fieldName) => {
     // If a field has been previously flagged as a lazy field, then
     // return true
-    const storedFields = typeFields.get(queryType.name)
+    const storedFields = typeFields.get(fieldType.name)
     if (storedFields && storedFields.has(fieldName)) {
       return true
     } else {
       // Otherwise, the filter field might be an array of linked
       // nodes, in which case we might filter via an elemMatch
       // field. Or, it might be a nested linked object. In either
       // case, we recurse
-      const gqlFieldType = fieldType.getFields()[fieldName].type
+      const gqlFieldType = fieldType.getFields()[fieldName]?.type
```

Comment:
the optional chaining here was needed because our filters end with operators object which is not part of output schema and it would cause crashes (trying to read `type` of undefined)